internal/artifactory_cache_test.go

```diff
@@ -0,0 +1,139 @@
+// artifactory_test.go
+package internal
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/cisco-open/grabit/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddWithArtifactoryCache(t *testing.T) {
+	t.Run("TokenNotSet", func(t *testing.T) {
+		os.Unsetenv("GRABIT_ARTIFACTORY_TOKEN")
+
+		handler := func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte(`test content`))
+		}
+		port, server := test.HttpHandler(handler)
+		defer server.Close()
+
+		path := test.TmpFile(t, "")
+		lock, err := NewLock(path, true)
+		require.NoError(t, err)
+
+		// Use a testing server
+		sourceURL := fmt.Sprintf("http://localhost:%d/test.txt", port)
+		cacheURL := fmt.Sprintf("http://localhost:%d", port)
+
+		err = lock.AddResource([]string{sourceURL}, "sha256", []string{}, "", cacheURL)
+		assert.Contains(t, err.Error(), "GRABIT_ARTIFACTORY_TOKEN environment variable is not set")
+	})
+}
+func TestDownloadWithArtifactoryCache(t *testing.T) {
+	t.Run("NO_CACHE_UPLOAD", func(t *testing.T) {
+		// Turn on NO_CACHE_UPLOAD setting
+		os.Setenv("NO_CACHE_UPLOAD", "1")
```

Comment:
Use https://beta.pkg.go.dev/testing@master#T.Setenv instead. This guarantees that it gets restored after the test runs.